(4b) Squads for VP/ThirdParty/PlotIterators.lua

```diff
@@ -0,0 +1,238 @@
+--
+-- Plot iterator functions
+--   All functions create an iterator centered on pPlot starting in the specified sector and working in the given direction around the ring/area
+--   PlotRingIterator(pPlot, r, sector, anticlock) returns each plot around a border of radius r
+--   PlotAreaSpiralIterator(pPlot, r, sector, anticlock, inwards, centre) returns each plot within an area of radius r in concentric rings
+--   PlotAreaSweepIterator(pPlot, r, sector, anticlock, inwards, centre) returns each plot within an area of radius r by radial
+--
+-- By converting the Map (x,y) co-ordinates to hex (x,y,z) co-ordinates we can simply walk the edges of the bounding hex
+-- To walk an edge we keep one of x, y or z invariant, iterate the others from 0 to r-1 and r-1 to 0  (r-1 otherwise we include each vertex twice)
+--
+-- By using Lua coroutines we can keep the walking code (fairly) simple
+--
+-- Use these functions as follows
+--
+--    for pEdgePlot in PlotRingIterator(pPlot, r, sector, anticlock) do
+--      print(pEdgePlot:GetX(), pEdgePlot:GetY())
+--    end
+--
+--    for pAreaPlot in PlotAreaSpiralIterator(pPlot, r, sector, anticlock, inwards, centre) do
+--      print(pAreaPlot:GetX(), pAreaPlot:GetY())
+--    end
+--
+--    for pAreaPlot in PlotAreaSweepIterator(pPlot, r, sector, anticlock, inwards, centre) do
+--      print(pAreaPlot:GetX(), pAreaPlot:GetY())
+--    end
+--
+
+SECTOR_NORTH = 1
+SECTOR_NORTHEAST = 2
+SECTOR_SOUTHEAST = 3
+SECTOR_SOUTH = 4
+SECTOR_SOUTHWEST = 5
+SECTOR_NORTHWEST = 6
+
+DIRECTION_CLOCKWISE = false
+DIRECTION_ANTICLOCKWISE = true
+
+DIRECTION_OUTWARDS = false
+DIRECTION_INWARDS = true
+
+CENTRE_INCLUDE = true
+CENTRE_EXCLUDE = false
+
+function PlotRingIterator(pPlot, r, sector, anticlock)
+  --print(string.format("PlotRingIterator((%i, %i), r=%i, s=%i, d=%s)", pPlot:GetX(), pPlot:GetY(), r, (sector or SECTOR_NORTH), (anticlock and "rev" or "fwd")))
+  -- The important thing to remember with hex-coordinates is that x+y+z = 0
+  -- so we never actually need to store z as we can always calculate it as -(x+y)
+  -- See http://keekerdc.com/2011/03/hexagon-grids-coordinate-systems-and-distance-calculations/
+
+  if (pPlot ~= nil and r > 0) then
+    local hex = ToHexFromGrid({x=pPlot:GetX(), y=pPlot:GetY()})
+    local x, y = hex.x, hex.y
+
+    -- Along the North edge of the hex (x-r, y+r, z) to (x, y+r, z-r)
```

Comment:
Can't these function be declared as also side in outer scope, looks like they're pur, so no need to reallocate them each time?